### PR TITLE
Added ModifySerializationForUnlock method

### PR DESCRIPTION
**Pull request by alexamiryan**
Added ModifySerializationForUnlock method for modifying existing Serialization for unlocking scenario if needed

### CODE REVIEW

**Review by Sajidur78 — CHANGES_REQUESTED** [review]
I suggest changing ``ModifySerializationForUnlock(byte[] Serialization)`` to ``ModifySerializationType(byte[] serialization, KerbLogoSubmitType type)`` and using the type argument to set the SubmitType.

**Comment by alexamiryan** [comment]
Maybe we can just change the method name? Actually it modifies Logon serialization to become a unlock serialization, so there is no point to giving it logon serialization and request logon serialization, it shouldn't modify it in that case

**Comment by Sajidur78** [comment]
Having a type argument in the function makes it easier to re-use in the future if it will ever be needed.

File: CredNet/CredentialSerializer.cs [hunk 1]
```diff
@@ -58,14 +58,16 @@ public static unsafe byte[] SerializeKerbWorkstationUnlock(string domain, string
...
-        public static unsafe byte[] ModifySerializationForUnlock(byte[] Serialization)
+        public static unsafe byte[] ModifySerializationForUnlock(byte[] Serialization, KerbLogonSubmitType type)
         {
-            fixed (byte* buffer = Serialization)
-            {
-                ((KerberosInteractiveLogon*) buffer)->SubmitType = KerbLogonSubmitType.WorkstationUnlockLogon;
+            if (type == KerbLogonSubmitType.WorkstationUnlockLogon) {
+                fixed (byte* buffer = Serialization)
+                {
+                    ((KerberosInteractiveLogon*)buffer)->SubmitType = KerbLogonSubmitType.WorkstationUnlockLogon;
```
Comment (Sajidur78) [inline CredNet/CredentialSerializer.cs]: Forgot to set the type here.

**Comment by alexamiryan** [comment]
Please check now

**Pull request merged**